src/feature/dirauth/process_descs.c

```diff
@@ -307,9 +306,11 @@ dirserv_router_get_status(const routerinfo_t *router, const char **msg,
   }
 
   /* Check for the more usual versions to reject a router first. */
-  uint32_t r = dirserv_get_status_impl(d, router->nickname, router->addr,
-                                       router->or_port, router->platform,
-                                       msg, severity);
+  uint32_t r = dirserv_get_status_impl(d,
+      router->cache_info.signing_key_cert->signing_key.pubkey,
```

Comment:
What if the router does not have an ed25519 key?
This code will crash.

Reply:
if (router->cache_info.signing_key_cert) {
    /* This has an ed25519 identity key. */

Reply:
b6465c0aec08febe5d40dd343838095cb96faf1d